### PR TITLE
PUBDEV-8937: fix sort OOM error on multi-node [nocheck]

### PR DESCRIPTION
**Pull request by wendycwong**
This PR fixes the issue in: https://h2oai.atlassian.net/browse/PUBDEV-8937
or in GH: https://github.com/h2oai/h2o-3/issues/6545

There are two things that can blow up the memory:
1. During the sort stage, one node will have to store the sorting cols and the absolute row number ordering after it has finished the sorting.
2. During the sorted frame generation process, one node will have to store batchsize * all columns in the frame.  

Hence, OOM can occur during either 1 or 2.  Hence, instead of fixing the batchsize, when the memory per node is small, we will try to reduce the batchsize to prevent OOM during 1 or 2.  When there is not enough memory in the nodes, instead of silently throwing an OOM in the java backend, we will throw an error to the clients and alert the users.

To determine what is the smallest size needed for each node to have in order to be able to do our sorting, I use a MEM_MULTIPLIER to determine how to scale up the memory needed.  From my experimentation, a value of 30 is needed in order to not have the OOM error.

### CODE REVIEW

**Review by tomasfryda — CHANGES_REQUESTED** [review]
Mostly, it looks good but there is one `Log.info` that I don't like.

File: h2o-core/src/main/java/water/rapids/RadixOrder.java [hunk 1]
```diff
@@ -53,12 +53,36 @@ public void compute2() {
...
+    // an array of size batchsize by numCols will be created for each sorted chunk in the end.  Memory is in bytes
+    long expandFactor = 8*((long) _DF.numCols())*MEM_MULTIPLIER; // 8 to translate 64 bits into 8 bytes, 100 is a scale up factor 
+    long batchMemory = Math.max((long) batchSize*expandFactor, minSortMemory);
+    if (batchMemory > minMem) {  // batchsize is too big for node with smallest memory, reduce it
+      batchSize = (int) Math.floor(minMem/expandFactor);
+      if (batchSize == 0)
+        throw new RuntimeException("The minimum memory of one of your nodes is too small to accommodate the sorting " +
+                "operation.  Make sure the smallest node has at least "+100*expandFactor+" bytes of memory.");
+    }
+    Log.info("batchsize ", batchSize);
```
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-core/src/main/java/water/rapids/RadixOrder.java:79]: I changed the error message.  Basically, the error message will tell the user how much memory they need in order to perform the sorting/merging action.  I do not want to go into too much detail and bring up the batchSize.  It is dependent on the size of their dataset and the number of columns that they want to sort/merge.

[the remark this answers is not in the document]
Comment (tomasfryda) [reply h2o-core/src/main/java/water/rapids/RadixOrder.java:79]: Sorry I wasn't clear, I meant the `Log.info("batchsize ", batchSize);` line. When we'll get some logs to debug the "batchsize xxx" doesn't seem that helpful to me as you'll have to know to what that refers to and possibly it's not that convenient to `grep 'batchsize' log` as the word is probably used often.

[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-core/src/main/java/water/rapids/RadixOrder.java:79]: That one is actually for my own use.  I will remove it then.

**Pull request merged**